internal/cli/atlas/deployments/options/deployment_opts_pre_run.go

```diff
@@ -0,0 +1,221 @@
+	"github.com/mongodb/mongodb-atlas-cli/internal/podman"
+	"github.com/shirou/gopsutil/v3/host"
+	"go.mongodb.org/atlas-sdk/v20230201008/admin"
+	"go.mongodb.org/atlas/mongodbatlas"
+)
+
+func (opts *DeploymentOpts) SelectDeployments(ctx context.Context, projectID string) error {
+	if opts.IsLocalDeploymentType() {
+		return opts.displayLocalDeployments(ctx)
+	}
+
+	if opts.IsAtlasDeploymentType() {
+		return opts.displayAtlasDeployments(projectID)
+	}
+
+	if !opts.IsCliAuthenticated() {
+		return opts.displayLocalDeployments(ctx)
+	}
+
+	var atlasDeployments []Deployment
+	var atlasErr error
+	if atlasDeployments, atlasErr = opts.GetAtlasDeployments(projectID); atlasErr != nil {
+		if err := opts.displayLocalDeployments(ctx); err != nil {
+			return err
+		}
+
+		_, _ = log.Warningf("Failed to retrieve Atlas deployments with: %s", atlasErr.Error())
+		return nil
+	}
+
+	if err := opts.LocalDeploymentPreRun(ctx); err != nil {
+		return err
+	}
+
+	localDeployments, err := opts.GetLocalDeployments(ctx)
+	if err != nil && !errors.Is(err, podman.ErrPodmanNotFound) {
+		return err
+	}
+
+	if opts.DeploymentName == "" {
+		return opts.Select(append(localDeployments, atlasDeployments...))
+	}
+
+	return opts.findDeployment(localDeployments, atlasDeployments)
+}
+
+func (opts *DeploymentOpts) findDeployment(localDeployments []Deployment, atlasDeployments []Deployment) error {
+	deployments := make([]Deployment, 0)
+	for _, d := range localDeployments {
+		if d.Name == opts.DeploymentName {
+			deployments = append(deployments, d)
+		}
+	}
+
+	for _, d := range atlasDeployments {
+		if d.Name == opts.DeploymentName {
+			deployments = append(deployments, d)
+		}
+	}
+
+	return opts.Select(deployments)
+}
+
+func (opts *DeploymentOpts) displayLocalDeployments(ctx context.Context) error {
+	if err := opts.LocalDeploymentPreRun(ctx); err != nil {
+		return err
+	}
+
+	localDeployments, err := opts.GetLocalDeployments(ctx)
+	if err != nil && !errors.Is(err, podman.ErrPodmanNotFound) {
+		return err
+	}
+
+	if opts.DeploymentName != "" {
+		for _, d := range localDeployments {
+			if d.Name == opts.DeploymentName {
+				opts.DeploymentType = LocalCluster
+				opts.DeploymentName = d.Name
+				return nil
+			}
+		}
+	}
+
+	return opts.Select(localDeployments)
+}
+
+func (opts *DeploymentOpts) displayAtlasDeployments(projectID string) error {
+	if !opts.IsCliAuthenticated() {
+		return ErrNotAuthenticated
+	}
+
+	var atlasDeployments []Deployment
+	var atlasErr error
+	if atlasDeployments, atlasErr = opts.GetAtlasDeployments(projectID); atlasErr != nil {
+		return atlasErr
+	}
+
+	if opts.DeploymentName != "" {
+		for _, d := range atlasDeployments {
+			if d.Name == opts.DeploymentName {
+				opts.DeploymentType = AtlasCluster
+				opts.DeploymentName = d.Name
+				return nil
+			}
+		}
+	}
+
+	return opts.Select(atlasDeployments)
+}
+
+func (opts *DeploymentOpts) GetAtlasDeployments(projectID string) ([]Deployment, error) {
```

Comment:
```suggestion
func (opts *DeploymentOpts) AtlasDeployments(projectID string) ([]Deployment, error) {
```
prefix get is not really idiomatic in Go